### PR TITLE
Use interpolated strings 6

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Replace `string.Format()` with new `string.Create((IFormatProvider? provider, [InterpolatedStringHandlerArgument(nameof(provider))] ref DefaultInterpolatedStringHandler handler)`.

## PR Context
#18974 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by iSazonov — CHANGES_REQUESTED** [review]
@CarloToso We should use
[StringBuilder.Append()](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs,1080) instead of StringBuilder.AppendFormat(). Earlier I pointed out to you the wrong method.

Please look `Use interpolated strings 8` and `Use interpolated strings 7` too.

File: src/System.Management.Automation/engine/GetCommandCommand.cs [hunk 1]
```diff
@@ -614,12 +614,7 @@ private PSObject GetSyntaxObject(CommandInfo command)
                 switch (command)
                 {
                     case ExternalScriptInfo externalScript:
-                        replacedSyntax = string.Format(
-                            "{0} (alias) -> {1}{2}{3}",
-                            aliasName,
-                            string.Format("{0}{1}", externalScript.Path, Environment.NewLine),
-                            Environment.NewLine,
-                            command.Syntax.Replace(command.Name, aliasName));
+                        replacedSyntax = string.Format($"{aliasName} (alias) -> {externalScript.Path}{Environment.NewLine}{Environment.NewLine}{command.Syntax.Replace(command.Name, aliasName)}");
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/GetCommandCommand.cs:617]: string.Format ?

File: src/System.Management.Automation/engine/GetCommandCommand.cs [hunk 2]
```diff
@@ -631,12 +626,7 @@ private PSObject GetSyntaxObject(CommandInfo command)
                         }
                         else
                         {
-                            replacedSyntax = string.Format(
-                                "{0} (alias) -> {1}{2}{3}",
-                                aliasName,
-                                command.Name,
-                                Environment.NewLine,
-                                command.Syntax.Replace(command.Name, aliasName));
+                            replacedSyntax = string.Format($"{aliasName} (alias) -> {command.Name}{Environment.NewLine}{command.Syntax.Replace(command.Name, aliasName)}");
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/GetCommandCommand.cs:629]: string.Format ?

Comment (CarloToso) [reply src/System.Management.Automation/engine/GetCommandCommand.cs:629]: String.Create doesn't accept only 1 argument, what format provider should I add?

Comment (iSazonov) [reply src/System.Management.Automation/engine/GetCommandCommand.cs:629]: Yes, we can add provider. I think Invariant is ok since it is strings only.

**Review by iSazonov — APPROVED** [review]

**Pull request merged**